Frameworks/Foundation/NSOrderedSet.mm

```diff
@@ -14,217 +14,297 @@
 //
 //******************************************************************************
 
-#include "Starboard.h"
+#import <Foundation/NSException.h>
+#import <NSOrderedSetInternal.h>
+#import <Starboard.h>
 #import <StubReturn.h>
+#import <VAListHelper.h>
 
-#include "Foundation/NSOrderedSet.h"
+// Needed to make sure that the returned array behaves like a one way proxy to an array
+@implementation _NSProxyOrderedSetArray
+- (instancetype)initWithOrderedSet:(NSOrderedSet*)orderedSet {
+    if (self = [super init]) {
+        _array = orderedSet->_arrayContainer;
```

Comment:
What you describe seems OK to me.

The problem I want to avoid is with code like this:

NSOrderedSet* oset = [NSOrderedSet new];
NSArray* array = [oset array];
[oset release];
[array count];  // Crash accessing deallocated object

---
In reply to: [64047960](https://github.com/Microsoft/WinObjC/pull/482#discussion_r64047960) [](ancestors = 64047960)